From 7e4192b70ed9d6e9ff37340cf9162eebdf8bb440 Mon Sep 17 00:00:00 2001
From: Ivan Romanov <drizt@land.ru>
Date: Wed, 28 Oct 2015 22:49:20 +0500
Subject: [PATCH 05/11] gnupg: resolve symbolic links

Keyrings watcher can gets paths in any format. To avoid possible
problems try to use unified names for paths.
---
 plugins/qca-gnupg/ringwatch.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/plugins/qca-gnupg/ringwatch.cpp b/plugins/qca-gnupg/ringwatch.cpp
index 263731d..96abf07 100644
--- a/plugins/qca-gnupg/ringwatch.cpp
+++ b/plugins/qca-gnupg/ringwatch.cpp
@@ -48,7 +48,10 @@ RingWatch::~RingWatch()
 void RingWatch::add(const QString &filePath)
 {
 	QFileInfo fi(filePath);
-	QString path = fi.absolutePath();
+	// Try to avoid symbolic links
+	QString path = fi.canonicalPath();
+	if (path.isEmpty())
+		path = fi.absolutePath();
 
 	// watching this path already?
 	DirWatch *dirWatch = 0;
-- 
2.5.0

